source/fem-dem/gls_sharp_navier_stokes.cc

```diff
@@ -2516,431 +2642,414 @@ GLSSharpNavierStokesSolver<dim>::sharp_edge()
-                      try
-                        {
-                          cell_2 = LetheGridTools::
-                            find_cell_around_point_with_neighbors<dim>(
-                              this->dof_handler,
-                              vertices_to_cell,
-                              cell,
-                              point_to_find_cell);
-                        }
-                      catch (...)
+                      const unsigned int component_i =
+                        this->fe->system_to_component_index(i).first;
+                      bool dof_is_inside =
+                        particles[ib_particle_id].get_levelset(
+                          support_points[local_dof_indices[i]], cell) <= 0;
+
+                      // If multiple particles cut the cell, we treat the dof of
+                      // pressure as a dummy dof. We don't use them to set the
+                      // boundary condition for the Poisson problem inside the
+                      // particle.
+                      bool use_ib_for_pressure =
+                        (dof_is_inside) && (component_i == dim) &&
+                        (this->simulation_parameters.particlesParameters
+                           ->assemble_navier_stokes_inside == false) &&
+                        count_particles < 2;
+
+                      // Check if the DOfs is owned and if it's not a hanging
+                      // node.
+                      if (((component_i < dim) || use_ib_for_pressure) &&
+                          this->locally_owned_dofs.is_element(
+                            global_index_overwrite) &&
+                          (this->zero_constraints.is_constrained(
+                             local_dof_indices[i]) ||
+                           this->nonzero_constraints.is_constrained(
+                             local_dof_indices[i])) == false)
                         {
-                          // If we are here, the DOF is on a boundary.
-                          particle_close_to_wall = true;
-                          cell_2                 = cell;
-                          // If a boundary condition is already applied to this
-                          // DOF we skip it otherwise we impose a value base on
-                          // the velocity of the particle.
-                          if (this->zero_constraints.is_constrained(
-                                global_index_overwrite) ||
-                              this->nonzero_constraints.is_constrained(
-                                global_index_overwrite))
+                          // We are working on the velocity of the cell cut
+                          // loops on the dof that are for vx or vy separately
+                          // loops on all the dof of the cell that represent
+                          // a specific component
+                          // Define which dof is going to be redefined
```

Comment:
This sentence is not clear. What is the velocity of the cell cut loops?
Why mention vx and vy? What about vz?